pkg/common/settings.go

```diff
@@ -0,0 +1,36 @@
+package common
+
+import "gopkg.in/yaml.v3"
+
+// EverestSettings represents the everest settings.
+type EverestSettings struct {
+	OIDC OIDCConfig
+}
+
+// OIDCConfig represents the OIDC provider configuration.
+type OIDCConfig struct {
+	IssuerURL string `yaml:"issuerUrl"`
+	ClientID  string `yaml:"clientId"`
+}
+
+const oidcMapKey = "oidc"
+
+// ToMap converts the EverestSettings struct to a map struct.
+func (e *EverestSettings) ToMap() (map[string]string, error) {
+	bytes, err := yaml.Marshal(e.OIDC)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]string)
+	result[oidcMapKey] = string(bytes)
+	return result, nil
+}
+
+// FromMap tries to convert a map the EverestSettings struct.
+func (e *EverestSettings) FromMap(m map[string]string) error {
+	err := yaml.Unmarshal([]byte(m[oidcMapKey]), &e.OIDC)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
I'm not really happy about this approach in compare with the flat list because:
1. The hardcoded `oidcMapKey` constant 
2. The fact that we would need to change the `ToMap()` and `FromMap()` methods each time we add a new setting.

The approach with the flat list doesn't have this downsides. Please let me know what approach do you prefer.

Reply:
Hey @oksana-grishchenko 

Thanks for trying this. I agree with your points. But out of curiosity, I took a look at how argocd handles this since we have the exact same use-case.

It seems that in their internal structures, they represent the OIDC config as a raw YAML string, and have additional methods to convert it to a struct and back to raw.

I tried to replicate their approach, and together with the `mapstructure` library, I think it tackles both the issues you mentioned. Here's an example: https://github.com/percona/everest/commit/092a06a805db2892dfb4607b0cffd64be4e37aea

Please let me know what you think :)

Reply:
Looks great, this approach is much more flexible, thank you Mayank! Cherry-picked your commit and updated the tests.